.pre-commit-config.yaml

```diff
@@ -55,7 +55,19 @@ repos:
       - id: make-spellcheck
         name: Run make verify-shellcheck
         entry: make verify-shellcheck
-        stages: [pre-push]
+        stages: [pre-commit]
+        language: system
+        require_serial: true
+      - id: make-verify-codespell
+        name: Run make verify-codespell
+        entry: make verify-codespell
+        stages: [ pre-commit ]
+        language: system
+        require_serial: true
```

Comment:
What do you say if we move `make verify-codespell` to `pre-push` stage ?

I would love to keep the `git commit` time to be as short as possible. 
Besides, we do run all the verify tests as part of a job. Reference conversation: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5213

Reply:
Sure, the removed a second off the time.

Reply:
Change has been pushed

Reply:
I tested `make verify-lint` in my local and it took the system `2m 24s` to complete it. 
What do you say to moving `verify-lint` to pre-push stage as well? Since verify-lint is not commit restricted, it should still check the linter-errors in complete changes. 

Sorry, not trying to be a bum about this 😶‍🌫️ I am quite inclined on having faster commit times.

Reply:
Wow 2m 🤯  

Sure I can move it to pre-push.

Reply:
Updated